rpc/v8/subscriptions.go

```diff
@@ -159,20 +166,24 @@ func (h *Handler) SubscribeEvents(ctx context.Context, fromAddr *felt.Felt, keys
 						return
 					}
 
-					for i, r := range block.Receipts {
-						for _, e := range r.Events {
-							fe := blockchain.FilteredEvent{
-								Event:           e,
-								BlockNumber:     &block.Number,
-								BlockHash:       header.Hash,
-								TransactionHash: block.Transactions[i].Hash(),
+					for _, r := range block.Receipts {
+						for i := range r.Events {
+							sentEvent := SentEvent{
+								TransactionHash: *r.TransactionHash,
+								EventIndex:      i,
 							}
-
-							delete(eventsPreviouslySent, fe)
+							delete(eventsPreviouslySent, sentEvent)
 						}
 					}
-				case pending := <-pendingSub.Recv():
-					h.processEvents(subscriptionCtx, w, id, pending.Number, pending.Number, fromAddr, keys, eventsPreviouslySent)
+				case <-pendingSub.Recv():
+					height, err := h.bcReader.Height()
```

Comment:
I think this can cause race condition, when the height increases after this call

Reply:
It can, it looks like that for me. One idea that comes to mind is that we can send the block directly to process event, instead of doing to separate requests 🤔.

Reply:
Resolving and creating #2516